tests/playwright/tests/checkout.spec.ts

```diff
@@ -0,0 +1,156 @@
+    "id": "",
+    "items": null,
+    "payment_status": null,
+    "reason": "upcoming",
+    "start_of_cycle": "2023-11-29T11:14:37Z",
+    "status": "draft",
+    "tax_amount": 9000,
+    "total": 54000,
+    "url": null })
+
+    await page.locator(':nth-match(:text("Edit"), 1)').click();
+    await page.getByLabel("Country/Region:").selectOption({ label: 'France' })
+    await page.locator(':nth-match(:text("Save"), 1)').click();
+    
+    await page.waitForTimeout(3000);
+
+    const country = await page.$('[data-testid="country"]')
+    const countryText = await country?.innerText();
+
+    expect(countryText).toBe("France")
+
+    await expect(page.locator('[data-testid="tax"]')).toBeVisible()
+    await expect(page.locator('[data-testid="total"]')).toBeVisible();
+  })
+})
+
+test.describe("Checkout - your inforamtion", ()=>{
+  test("user: clicks cancel should reset field", async ({page}) => {
+    await page.goto("/pro/subscribe")
+    await acceptCookiePolicy(page)
+    await selectProducts(page);
+    await page.getByRole("button", { name: "Buy now" }).click();
+    
+    await page.click('button[type="submit"]')
+    await expect(page).toHaveURL('/account/checkout');
+    
+    await page.locator(':nth-match(:text("Edit"), 2)').click();
+    await page.getByLabel("Name:").fill("Abcd")
+    await page.getByLabel("Address:").fill("Abcd");
+    await page.getByLabel("City:").fill("Abcd");
+    await page.getByLabel("Postal code:").fill("Abcd");
+    await page.locator(':nth-match(:text("Save"), 1)').click();
+
+    expect(await page.$("data-testid=customer-name")).toBeNull();
+    expect(await page.$("data-testid=customer-address")).toBeNull();
+    expect(await page.$("data-testid=customer-city")).toBeNull();;
+    expect(await page.$("data-testid=customer-postal-code")).toBeNull();
+  })
+})
+
+test.describe("Checkout purchase", ()=>{
+  test("user: it should purchase", async ({page}) => {
+    await page.goto("/pro/subscribe")
+    await acceptCookiePolicy(page)
+    await selectProducts(page);
+    await page.getByRole("button", { name: "Buy now" }).click();
+    
+    await page.click('button[type="submit"]')
+    await expect(page).toHaveURL('/account/checkout');
+    
+    await acceptTerms(page)
+    await clickRecaptcha(page)
+    await page.waitForTimeout(3000)
+
+    await page.route(ENDPOINTS.preview, async (route) => {
+      await route.fulfill({
+        status: 200,
+        body: JSON.stringify({ currency: "USD",
+        subtotal: 16700,
+        tax: 1670,
+        total: 18370 })
+      })
+    })
+    await page.route(ENDPOINTS.postPurchase, async (route) => {
+      await route.fulfill({
+        status: 200,
+        body: JSON.stringify({
+          account_id: "test",
+          marketplace: "canonical-ua",
+          action: "purchase",
+          previous_purchase_id: "test"
+        })
+      });
+    })
+      await page.route(ENDPOINTS.getPurchase, async(route) => {
```

Comment:
According to this [doc](https://playwright.dev/docs/mock), this is how to intercept the calls and make mock apis. 
```
page.route( "url", async ( route ) => {
     await route.fulfill({ json })
  });
}
```
First, I'm not sure if this is right way of mocking apis. What I understand is when the api call that includes "url" is made, the code above will intercept the call and return the "json" part instead of making a real api request and response.  

I added the return values of the api calls with random value. wasn't sure if this values are okay to add and what to "expect"  after.